File: BE_QuanLyTro/src/main/java/ut/edu/be_quanlytro/Dto/Response/InvoiceResponse.java
package ut.edu.be_quanlytro.Dto.Response;

import lombok.Builder;
import lombok.Data;
import java.math.BigDecimal;
import java.time.LocalDate;
import java.util.UUID;

@Data
@Builder
public class InvoiceResponse {
    private UUID id;
    private String roomNumber;
    private LocalDate invoicePeriod;
    private LocalDate dueDate;
    private BigDecimal roomPrice; // Tiền phòng gốc
    private BigDecimal totalAmount; // Tổng tiền (Phòng + Điện + Nước + Rác...)
    private String status;
}

File: Flutter_QuanLyTro/lib/data/models/response/invoice_response.dart
class InvoiceResponse {
  final String id;
  final String roomNumber;
  final String invoicePeriod;
  final String dueDate;
  final double roomPrice;
  final double totalAmount;
  final String status;

  InvoiceResponse({
    required this.id,
    required this.roomNumber,
    required this.invoicePeriod,
    required this.dueDate,
    required this.roomPrice,
    required this.totalAmount,
    required this.status,
  });

  factory InvoiceResponse.fromJson(Map<String, dynamic> json) {
    return InvoiceResponse(
      id: json['id'] ?? '',
      roomNumber: json['roomNumber'] ?? '',
      invoicePeriod: json['invoicePeriod'] ?? '',
      dueDate: json['dueDate'] ?? '',
      roomPrice: (json['roomPrice'] as num?)?.toDouble() ?? 0.0,
      totalAmount: (json['totalAmount'] as num?)?.toDouble() ?? 0.0,
      status: json['status'] ?? 'UNPAID',
    );
  }
}